mlflow/types/type_hints.py

```diff
@@ -262,21 +272,26 @@ def _get_element_type_of_list_type_hint(type_hint: type[list[Any]]) -> Any:
     Get the element type of list[...] type hint
     """
     args = get_args(type_hint)
+    # Optional[list[...]]
+    if type(None) in args:
+        raise MlflowException.invalid_parameter_value(OPTIONAL_INPUT_MSG)
     # a valid list[...] type hint must only contain one argument
     if len(args) == 0:
         raise MlflowException.invalid_parameter_value(
-            f"List type hint must contain the element type, got {type_hint}"
+            f"Type hint `{_type_hint_repr(type_hint)}` doesn't contain element type, fix it "
+            "by adding an element type, e.g. `list[str]` instead of `list`."
```

Comment:
```suggestion
            f"Type hint `{_type_hint_repr(type_hint)}` doesn't contain a collection element type. Fix "
            "by adding an element type to the collection type definition, e.g. `list[str]` instead of `list`."
```